code/parachain/runtime/common/src/lib.rs

```diff
@@ -170,24 +171,21 @@ impl<AssetsRegistry: AssetRatioInspect<AssetId = CurrencyId>> MinimalOracle
 			CurrencyId::PICA => Ok(amount),
 			_ =>
 				if let Some(ratio) = AssetsRegistry::get_ratio(asset_id) {
-					if let Some(amount) = Ratio::checked_from_integer(amount) {
-						if let Some(payment) = ratio.checked_mul(&amount) {
-							payment.into_inner().safe_div(&Ratio::accuracy()).map_err(Into::into)
-						} else {
-							Err(DispatchError::Other(
-								cross_chain_errors::AMOUNT_OF_ASSET_IS_MORE_THAN_MAX_POSSIBLE,
-							))
-						}
+					let amount = Ratio::from_inner(amount);
+					if let Some(payment) = ratio.checked_mul(&amount) {
+						Ok(payment.into_inner())
 					} else {
 						Err(DispatchError::Other(
 							cross_chain_errors::AMOUNT_OF_ASSET_IS_MORE_THAN_MAX_POSSIBLE,
 						))
 					}
-				// TODO: waiting values from product
+				// hardcoded assets -> hardcoded initial prices
 				} else if asset_id == CurrencyId::KSM {
-					Ok(amount / 123)
+					Ok(amount / 2667)
 				} else if asset_id == CurrencyId::kUSD {
-					Ok(amount / 13)
+					Ok(amount / 67)
+				} else if asset_id == CurrencyId::USDT || asset_id == CurrencyId::USDC {
+					Ok(amount * 1_000_000 / 67_000_000_000_000)
```

Comment:
Preform hard-coded checks before doing storage reads?

Reply:
Sudo can use AR to change route. Sure can prevent it from doing so. I do not know what is better. Better what is consistent with all other places.